app/javascript/controllers/rainy_day_controller.js

```diff
@@ -44,63 +110,140 @@ export default class extends Controller {
 //   var areaId = $target.data('id');
 //   var $selectedItems = $('[data-id="' + $target.data('id') + '"]');
 
 //   // Do nothing if the clicked element is already active
-//   if ($target.hasClass(this.options.activeSelectorClass))
-//   {
-//       return;
-//   }
-
+//   if ($target.hasClass(this.options.activeSelectorClass)) //activeSelectorClass = active{   
+//   return;}
 //   this.fetchArea(areaId)
 //       .then(this.displayArea.bind(this, $selectedItems));
 // }
 
-// RainyDayController.prototype.fetchArea = function (id) {
-//   var url = '/' + this.watchedAreaSlug + '/rainy-day-options/';
-//   return $.get(url + id);
-// }
+/****************** New method for handling area selection (NOT CURRENTlY USED) ************/
+handleAreaSelection(event) {
+  // let target = document.querySelector(event)
+  let areaId = event
+  let element = document.querySelectorAll(`[data-id="${areaId}"]`)
 
-// RainyDayController.prototype.displayArea = function ($newActiveListItem, response) {
-//   if (!response || response.length == 0) {
-//       console.error('Issue connecting to server :(');
-//   }
+  if (this.areaTarget) {
+    return
+  }
+  this.fetchArea(areaId)
+    .then(this.displayArea(() => {
+      element
+    }))
+  }
 
-//   this.swapActiveClass($newActiveListItem);
-//   this.updatePageInformation(response);
-//   this.updateMountainProjectWidget($newActiveListItem);
+/*********Used this function to make an async function for getting url***********/
+    // RainyDayController.prototype.fetchArea = function (id) {
+    //   var url = '/' + this.watchedAreaSlug + '/rainy-day-options/';
+    //   return $.get(url + id);
+    // }
+    fetchArea = async function(event) {
+      const href = this.linkTarget.getAttribute("href")
+      let url = href + '/rainy-day-options/';
+
+    const response = await fetch(url + event.currentTarget.dataset.id)
+        if (!response.ok) {
+          throw new Error('Network response error')
+        }
+        const data = await response.json()
+        
+        // We need to select the attributes from fetch
+        const description = data.climbing_area.description
+        const rockType = data.climbing_area.rock_type
+        // const drivingTime = data.driving_time
+
+
+        // We need to select the elements/attributes to be replaced from html
+        const areaDescription = document.querySelector('p[data-role="area-description"]')
+        const rockTypeElement = document.querySelector('strong[data-role="area-rock-type"]')
+        // const drivingTimeElement = document.querySelector('strong[data-role="area-drive-time"]')
+
+        // Use the removeHTMLTags function on the texts
+        const descriptionTagsRemoved = removeHTMLTags(description)
+        const rockTypeTagsRemoved = removeHTMLTags(rockType)
+        const areaDescriptionTagsRemoved = removeHTMLTags(areaDescription.innerText)
+        const rockTypeElementTagsRemoved = removeHTMLTags(rockTypeElement.innerText)
```

Comment:
Removing the HTML tags here causes the styling on these sections to be lost, giving us a big ol blob of text that isn't very readable:

![Screenshot 2024-10-27 at 3 20 59 PM](https://github.com/user-attachments/assets/ebb58374-b413-4f49-a18f-9b6b3e112636)

Instead of removing the HTML, let's instead replace the content of the tags with exactly what we're receiving. `displayArea` below does this but it looks like you aren't using this method, so we should either call that method here or perform that logic here instead.

Reply:
Done!